packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -1311,8 +1317,12 @@ class _CupertinoActionSheetActionState extends State<CupertinoActionSheetAction>
     final double verticalPadding = _kActionSheetButtonVerticalPaddingBase
         + fontSize * _kActionSheetButtonVerticalPaddingFactor;
 
+    final Set<WidgetState> states = <WidgetState>{
+      WidgetState.hovered
+    };
+
     return MouseRegion(
-      cursor: kIsWeb ? SystemMouseCursors.click : MouseCursor.defer,
+      cursor: kIsWeb ? SystemMouseCursors.click : widget.mouseCursor,
```

Comment:
What if a user wants to configure a mouse cursor on web (instead of just click)? The user provided property should always take priority.

Reply:
want a suggestion if I can do the following:
`final MouseCursor webCursor = (kIsWeb && widget.mouseCursor != MouseCursor.defer) ? 
    widget.mouseCursor : SystemMouseCursors.click;`

Since mouseCursor is defaulted to MouseCursor.defer when its web and the mouseCursor is not equal to MouseCursor.defer then we use the cursor passed by the user else we default to SystemMouseCursor.click

Reply:
Instead of making `mouseCursor` default to `MouseCursor.defer` in the constructor, you can make it nullable. Then when setting the `MouseRegion`s cursor, have `mouseCursor` null coalesce to either `.click` on web or `.defer` otherwise.